### PR TITLE
Create endpoint to get all user's favorite events

### LINKED ISSUE #6209: [GreenCity] Create endpoint to gell all user's favorite events
Create endpoint to gell all user's favorite events

### PR DESCRIPTION
**Pull request by Yurii-Midianyi**
Issue link [6209](https://github.com/ita-social-projects/GreenCity/issues/6209)

### CODE REVIEW

File: core/src/main/java/greencity/config/SecurityConfig.java [hunk 1]
```diff
@@ -153,6 +153,7 @@ protected void configure(HttpSecurity http) throws Exception {
                 EVENTS,
                 EVENTS + "/event/{eventId}",
                 EVENTS + "/getAllSubscribers/{eventId}",
+                EVENTS + "/getAllFavoriteEvents",
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/config/SecurityConfig.java]: how can unauthorized users have favorite events?

Comment (RomaMocherniuk) [reply core/src/main/java/greencity/config/SecurityConfig.java]: we should add the `userId` to the request or return the favorite events for authorized users only

**Review by RomaMocherniuk — CHANGES_REQUESTED** [review]
Please fix the comments

File: core/src/main/java/greencity/controller/EventsController.java [hunk 2]
```diff
@@ -169,6 +169,25 @@ public ResponseEntity<PageableAdvancedDto<EventDto>> getUserEvents(
...
+     * Method for getting all user's favorite events.
+     *
+     * @return a set of {@link EventDto} instance.
+     * @author Midianyi Yurii
+     */
+    @ApiOperation(value = "Get all user's favorite events")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = HttpStatuses.OK),
+        @ApiResponse(code = 400, message = HttpStatuses.BAD_REQUEST),
+        @ApiResponse(code = 404, message = HttpStatuses.NOT_FOUND)
```
Comment (RomaMocherniuk) [inline core/src/main/java/greencity/controller/EventsController.java:183]: please add `401` response code

File: dao/src/main/java/greencity/repository/EventRepo.java [hunk 3]
```diff
@@ -90,4 +89,12 @@ public interface EventRepo extends JpaRepository<Event, Long>, JpaSpecificationE
...
      */
     List<Event> getAllByOrganizer(User organizer);
+
+    /**
+     * Get all user's favorite events by user id.
+     *
+     * @param userId {@link Long}.
+     */
+    @Query(value = "SELECT e FROM Event e LEFT JOIN e.followers AS f WHERE f.id = :userId")
+    List<Event> findAllFavoritesByUser(Long userId);
```
Comment (RomaMocherniuk) [inline dao/src/main/java/greencity/repository/EventRepo.java]: can we return  `Page<Event>` here?

File: service-api/src/main/java/greencity/service/EventService.java [hunk 4]
```diff
@@ -141,4 +140,12 @@ PageableAdvancedDto<EventDto> getAllUserEvents(Pageable page, String email, Stri
      * @param eventId - id of event
      */
     EventVO findById(Long eventId);
+
+    /**
+     * Method for getting all user's favorite events.
+     * 
+     * @return a page of{@link EventDto} instance.
+     * @author Midianyi Yurii.
+     */
```
Comment (RomaMocherniuk) [inline service-api/src/main/java/greencity/service/EventService.java:151]: please add `Pageable pageable and String email `to Javadoc

**Review by RomaMocherniuk — APPROVED** [review]

**Review by ABbondar — APPROVED** [review]

**Review by sergoliarnik — APPROVED** [review]

**Review by 2or5 — APPROVED** [review]

**Pull request merged**